jmx-scraper/src/main/java/io/opentelemetry/contrib/jmxscraper/JmxScraper.java

```diff
@@ -104,52 +96,42 @@ private static void loadPropertiesFromPath(Properties props, String path)
   }
 
   JmxScraper(JmxScraperConfig config) throws ConfigurationException {
-    this.config = config;
 
-    try {
-      @SuppressWarnings("unused") // TODO: Temporary
-      JmxClient jmxClient = new JmxClient(config);
-    } catch (MalformedURLException e) {
-      throw new ConfigurationException("Malformed serviceUrl: ", e);
+    String serviceUrl = config.getServiceUrl();
+    if (serviceUrl == null) {
+      throw new ConfigurationException("missing service URL");
     }
+    int interval = config.getIntervalMilliseconds();
+    if (interval < 0) {
+      throw new ConfigurationException("interval must be positive");
+    }
+    this.client = JmxRemoteClient.createNew(serviceUrl);
+    // TODO: depend on instrumentation 2.9.0 snapshot
+    // this.service = JmxMetricInsight.createService(GlobalOpenTelemetry.get(), interval);
   }
 
-  @SuppressWarnings("FutureReturnValueIgnored") // TODO: Temporary
   private void start() {
-    exec.scheduleWithFixedDelay(
-        () -> {
-          logger.fine("JMX scraping triggered");
-          //            try {
-          //              runner.run();
-          //            } catch (Throwable e) {
-          //              logger.log(Level.SEVERE, "Error gathering JMX metrics", e);
-          //            }
-        },
-        0,
-        config.getIntervalMilliseconds(),
-        TimeUnit.MILLISECONDS);
+    @SuppressWarnings("unused")
+    MBeanServerConnection connection;
+    try {
+      JMXConnector connector = client.connect();
+      connection = connector.getMBeanServerConnection();
+    } catch (IOException e) {
+      throw new IllegalStateException(e);
```

Comment:
It might be useful in the future to log some addiitonal info here

Reply:
Good point, I will even add that we should probably propagate the `IOException` here so the caller can properly handle this (in this case the `main` method).